File: numeric/primes500/primes.rb

def prime_sieve()
    Enumerator.new do |yielder|
        yielder << 2
        cache = [2]
        candidate = 3
        while true
            if cache.all? { |prime| candidate % prime != 0 }
                yielder << candidate
                cache << candidate
            end

            candidate += 2
        end
    end
end

def get_primes(count)
    return prime_sieve().take(count)
end

def print_primes(primes)
    puts "First Five Hundred Primes"
    for i in 0..49 do
        puts "     %04d %04d %04d %04d %04d %04d %04d %04d %04d %04d" % [
            primes[i], primes[50 + i], primes[100 + i], primes[150 + i],
            primes[200 + i], primes[250 + i], primes[300 + i],
            primes[350 + i], primes[400 + i], primes[450 + i]]
    end
end

print_primes(get_primes(500))

File: numeric/primes500/primes.cs

static IEnumerable<int> PrimeSieve()
{
    yield return 2;
    List<int> cache = [2];
    int candidate = 3;
    while (true)
    {
        if (cache.All(prime => candidate % prime != 0))
        {
            yield return candidate;
            cache.Add(candidate);
        }

        candidate += 2;
    }
}

static List<int> GetPrimes(int count)
{
    return [.. PrimeSieve().Take(count)];
}

static void PrintPrimes(List<int> primes)
{
    Console.WriteLine("First Five Hundred Primes");
    for (int i = 0; i < 50; ++i)
    {
        Console.WriteLine($"     {primes[i]:0000} {primes[i + 50]:0000}" +
            $" {primes[i + 100]:0000} {primes[i + 150]:0000}" +
            $" {primes[i + 200]:0000} {primes[i + 250]:0000}" +
            $" {primes[i + 300]:0000} {primes[i + 350]:0000}" +
            $" {primes[i + 400]:0000} {primes[i + 450]:0000}");
    }
}

PrintPrimes(GetPrimes(500));
